onmt/translate/translation_server.py

```diff
@@ -518,43 +530,43 @@ def run(self, inputs):
                 tok = self.maybe_tokenize(seg)
                 if ref is not None:
                     ref = self.maybe_tokenize(ref, side='tgt')
-                inferred_feats = self.transform_feats(seg, tok, feats)
-                texts.append((tok, ref, inferred_feats))
+                feats = self.maybe_transform_feats(seg, tok, feats)
+                texts.append((tok, ref, feats))
             tail_spaces.append(whitespaces_after)
 
         empty_indices = []
-        texts_to_translate, texts_ref = [], []
-        texts_features = defaultdict(list)
+        examples = []
         for i, (tok, ref_tok, feats) in enumerate(texts):
             if tok == "":
                 empty_indices.append(i)
             else:
-                texts_to_translate.append(tok)
-                texts_ref.append(ref_tok)
-                for feat_name, feat_values in feats.items():
-                    texts_features[feat_name].append(feat_values)
-        if any([item is None for item in texts_ref]):
-            texts_ref = None
+                ex = {
+                    "src": {"src": tok},
+                    "tgt": {"tgt": ref_tok} if ref_tok is not None else None
+                }
+                if feats is not None:
+                    ex["src"]["feats"] = feats
+                examples.append(ex)
 
         scores = []
         predictions = []
-
-        if len(texts_to_translate) > 0:
+        if len(examples) > 0:
             try:
                 if isinstance(self.translator, CTranslate2Translator):
+                    # TODO
                     scores, predictions = self.translator.translate(
-                        texts_to_translate)
+                        examples)  # texts_to_translate)
```

Comment:
Remove unnecessary comment?

Reply:
Answered above. WIP

Reply:
I think we just translate without features with ctranslate2. So you need to use translate_batch and remove the unnecessary comments.